kubetools/config.py

```diff
@@ -151,8 +151,9 @@ def _check_min_version(config):
 
 def _filter_config_data(key, items_or_object, env, namespace, dev):
     def is_match(item):
+        conditions = None if 'batch/v1' in item else item.get('conditions')
         return _conditions_match(
-            item.get('conditions'),
+            conditions,
             env=env,
             namespace=namespace,
             dev=dev,
```

Comment:
What we want to have:
```
cronjobs:
  **batch-api-version: 'batch/v1beta1'   # Must add if k8s version < 1.21+**
  my-cronjob:
    schedule: "*/1 * * * *"
....
```

If I don't add this change, it tries to run `item.get('conditions')` on `'batch/v1beta1'` and it throws the following exception `--> Unexpected exception: AttributeError("'str' object has no attribute 'get'",)`

This function will return the filtered config, and we need `batch-api-version: ..` in it.
The function `_conditions_match` in `is_match(item)` return True or False to keep the config (or not) in the final return. So yes, it's a small hack to force the function to keep `batch-api-version: ..` in the final return so I can access this value [here](https://github.com/EDITD/kubetools/blob/d79ae78dcdf17ba3e3f103caecc8bd705a594a94/kubetools/kubernetes/config/cronjob.py#L23-L25)

Another thing that I can do, is to remove this change and modify [this part](https://github.com/EDITD/kubetools/blob/d79ae78dcdf17ba3e3f103caecc8bd705a594a94/kubetools/config.py#L168-L173):
```

    elif isinstance(items_or_object, dict):
        return {
            key: item
            for key, item in items_or_object.items()
            **if 'batch/v1' in item or is_match(item)**
        }
```

Reply:
Looks like the abstractions don't match then: `cronjobs` expects a dict of `job_name` -> `job_config`, so there is no place in there for a global setting applying to all cronjobs.

* Either we need to make a place for that config, but that would mean choosing a special key for the config, like you tried, in a place where all keys are valid and equivalent. This means a special handling of that key (with an explicit check on the key, not on the value), which is what your last suggestion is. Another place to put this config, instead of under `cronjobs`, would be at the top-level of the configuration file, ideally under a new section (`options` ? `settings` and `config` names are already taken)
* Or we reunite the abstractions by moving the config to a place where we can. So far the only place I can see is within each `cronjob`, for example:
  ```
  ...
  cronjobs:
      my-cronjob:
        batch-api-version: 'batch/v1beta1'   # Must add if k8s version < 1.21+
        schedule: "*/1 * * * *"
  ....
  ```
  Not ideal because it means repeating the setting for each cronjob. But if it should make the handling code simpler.